### ISSUE TITLE
index.html in the root is treated differently

### ISSUE THREAD

**Issue by eps1lon** [codesandbox/codesandbox-client#4982]
# 🐛 bug report

## Preflight Checklist

- [x] I have read the
      [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md)
      for this project.
- [x] I agree to follow the
      [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md)
      that this project adheres to.
- [x] I have searched the issue tracker for an issue that matches the one I want
      to file, without success.
  Similar to https://github.com/codesandbox/codesandbox-client/issues/4737

## Description of the problem

If I move `index.html` in the root instead of `public` the `<head />` tag is ignored. It is only included if I move `index.html` into `public/index.html`

## How has this issue affected you? What are you trying to accomplish?

Programmatically creating a sandbox with the API and a custom name (in the `<title />`). We'd like to keep the file structure flat but it isn't really a blocker right now.

### To Reproduce

Move `./public/index.html` into `./index.html`

### Your Environment

| Software         | Name/Version |
| ---------------- | ------------ |
| Сodesandbox      | 687f9a330
| Browser          | Chrome 85
| Operating System | Ubuntu 18.04

**Comment by SaraVieira** [comment]
Hey!

That's just how the create react app template works, I download actual CRA and when moving the index.html it errors out, in our case we use a default one so you sandbox doesn't break but for it to be recognized it needs to be in public.

I will close as this is expected behavior in Create React App and it does work in vanilla templates as you can see here: https://codesandbox.io/s/young-sound-24q28?file=/src/index.js

CRA simply expects the file to be there.

**Comment by sannek** [comment]
Hi,

Thanks for opening an issue. This is a bit of a weird one, because it's not really a bug so much as a bit of a quirk (because we tried to remain backwards compatible).

Because we want sandboxes to work locally as well, we use `react-scripts` as the CLI for React sandboxes, and `react-scripts` only works with `index.html` in `/public`, and not in the root. 

And this is where it gets a bit quirky: if we (CodeSandbox) were being really strict about it, your project shouldn't work _at all_ when you put `index.html` in the root. However, if we actually enforced that, we'd break a large number of old sandboxes, so right now your project will work on CodeSandbox (but not 100%) if you have `index.html` in the root, but outside of CodeSandbox it would not work at all.

In short, it's not a real option to put `index.html` in the root, so I'd encourage you to keep it in `/public` to make sure your project works as expected.

(Oh - Sara already explained this to you, but I already typed all this out so here's some more detail)

**Comment by eps1lon** [comment]
@SaraVieira Please look at https://github.com/codesandbox/codesandbox-client/issues/4737. It's not about CRA expecting the file to be there. Codesandbox treats it differently. It should either fail (like create-react-app) or support it in the root.

Failing silently is the worst option. Also note that not supporting index.html in the root would be a breaking change and really bad for all the deployed documentations that create programmatic codesandboxes in the root.

**Comment by SaraVieira** [comment]
Because we give you a generic file so it doesn't break 

We are not gonna break the ones that have it because of the "not breaking the web" principle, old sandboxes should never break but this is done on porpuse to be in touch with CRA

The way it works may not be the exact same way that CRA works but it's a middle ground to not break the web and still enforce the practices
